server_modules/file_system.js

```diff
@@ -5,8 +5,8 @@ var path = require('path')
 var utils = require('./utils');
 var args = require('./args');
 
-function absoluteFile(file) {
-    return (file[0] === '/' ? utils.root : '') + file;
+function absolutePath(file) {
+    return path.resolve(path.dirname(process.argv[1]), file);
 }
 
 function getDataFilePath(file) {
```

Comment:
This function could now use `absolutePath()`, i.e. `return absolutePath(path.join(args.datadir, file))`;